### PR TITLE
refactor: extract delete, extend, download workflows and published-ad fetch into focused modules

### PR DESCRIPTION
**Pull request by 1cu**
## ℹ️ Description

> **Why one PR?** The four extractions are mechanically identical: extract a method body into a module function, replace `self.*` with explicit parameters, update call sites in `run()`. Splitting into 4 sequential PRs would produce repetitive diffs against the same `__init__.py`, compound rebase friction, and add no independent review signal — each PR would be reviewed as "same pattern, different function." Combined, the mechanical nature is visible at a glance and the 17 files are mostly new modules + focused test files.

Reduce KleinanzeigenBot.__init__.py from 2865 to 2391 lines by extracting four browser workflow modules with explicit dependencies.

## 📋 Changes Summary

- New `published_ads.py` — `fetch_published_ads()` and `PublishedAdsFetchIncompleteError`, shared by all workflows
- New `delete_flow.py` — `delete_ads()` orchestrator and `delete_ad()` single-ad deletion
- New `extend_flow.py` — `extend_ads()` orchestrator and `_extend_ad()` single-ad extension
- New `download_flow.py` — `download_ads()`, `resolve_download_dir()`, and `_download_ad_with_resolved_state()`
- Promote `_timeout` and `_navigate_paginated_ad_overview` to public in WebScrapingMixin
- Restore `PublishedAdsFetchIncompleteError` base class to KleinanzeigenBotError
- New focused test files: `test_published_ads.py` (17), `test_delete_flow.py` (13), `test_extend_flow.py` (17), `test_download_flow.py` (16)
- Removed: `test_json_pagination.py`, `test_extend_command.py`
- Flow modules receive explicit dependencies (`web`, `root_url`, `config`, `workspace`, `load_ads_func`) — no `do_work(bot)` signatures
- Temporary `_fetch_published_ads` delegator kept on bot (still used by publish/update)

### ⚙️ Type of Change
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [x] ✨ New feature (adds new functionality without breaking existing usage)
- [ ] 💥 Breaking change (changes that might break existing user setups, scripts, or configurations)

## ✅ Checklist
- [x] I have reviewed my changes to ensure they meet the project's standards.
- [x] I have tested my changes and ensured that all tests pass  (`pdm run test`).
- [x] I have formatted the code (`pdm run format`).
- [x] I have verified that linting passes (`pdm run lint`).
- [x] I have updated documentation where necessary.

## Summary by CodeRabbit

* **Documentation**
  * Clarified timeout configuration guidance and examples (use public timeout API and when to use effective timeout).

* **New Features**
  * Added workflows to delete, download (all/new/IDs) and extend ads; added API-backed published-ads retrieval and safe ID matching.

* **Refactor**
  * Command routing moved to workflow modules; unified timeout and pagination helpers for consistent UI/navigation behavior.

* **Tests**
  * Expanded/updated unit tests covering deletion, download, extension, pagination, timeout and related edge cases.

### CODE REVIEW

File: src/kleinanzeigen_bot/delete_flow.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+from . import ad_state as _ad_state
+from . import published_ads
+from .model.ad_model import Ad
+from .utils import dicts as _dicts
+from .utils import loggers as _loggers
+from .utils.i18n import pluralize
+from .utils.misc import ensure
+from .utils.web_scraping_mixin import By, WebScrapingMixin
+
```
[the remark this answers is not in the document]
Comment (1cu) [reply src/kleinanzeigen_bot/delete_flow.py:18]: Deferred: Sibling modules in this repo (`ad_loading.py`, `ad_state.py`, etc.) consistently use relative imports for internal dependencies. Switching one module now would be inconsistent with the existing codebase pattern. This should be decided repo-wide, not per-file. Tracking separately.

File: src/kleinanzeigen_bot/published_ads.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+                preview = preview[:SNIPPET_LIMIT] + "..."
+            LOG.warning("Filtered %s malformed ad entries on page %s (sample: %s)", rejected_count, page, preview)
+            _handle_incomplete_fetch("Filtered %s malformed ad entries on page %s (sample: %s)", rejected_count, page, preview)
+
+        ads.extend(filtered_page_ads)
+
+        paging = json_data.get("paging")
+        if not isinstance(paging, dict):
+            LOG.debug("No paging dict found on page %s, assuming single page", page)
+            break
```
[the remark this answers is not in the document]
Comment (1cu) [reply src/kleinanzeigen_bot/published_ads.py:149]: Deferred: This is pre-existing behavior moved from `__init__.py`. The strict-mode contract for missing paging has been this way since before extraction. Changing it would alter behavior and needs its own tracked issue with test coverage — too risky as scope creep in a refactor PR.

File: src/kleinanzeigen_bot/published_ads.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+                break
+            snippet = content[:SNIPPET_LIMIT] + ("..." if len(content) > SNIPPET_LIMIT else "")
+            LOG.warning("Failed to parse JSON response on page %s: %s (content: %s)", page, ex, snippet)
+            _handle_incomplete_fetch("Failed to parse JSON response on page %s: %s (content: %s)", page, ex, snippet, cause = ex)
+            break
+
+        if not isinstance(json_data, dict):
+            snippet = content[:SNIPPET_LIMIT] + ("..." if len(content) > SNIPPET_LIMIT else "")
+            LOG.warning("Unexpected JSON payload on page %s (content: %s)", page, snippet)
+            _handle_incomplete_fetch("Unexpected JSON payload on page %s (content: %s)", page, snippet)
```
[the remark this answers is not in the document]
Comment (1cu) [reply src/kleinanzeigen_bot/published_ads.py:113]: Deferred: This is pre-existing behavior moved from `__init__.py`. The snippet logging for JSON parse failures existed before extraction. Changing log output from error payloads deserves deliberate security review in a tracked issue, not a silent change embedded in a workflow extraction. Tracking separately.
